scalalib/worker/src/mill/scalalib/worker/MockedLookup.scala

```diff
@@ -13,6 +13,9 @@ case class MockedLookup(am: VirtualFile => Optional[CompileAnalysis])
   override def analysis(classpathEntry: VirtualFile): Optional[CompileAnalysis] =
     am(classpathEntry)
 
-  override def definesClass(classpathEntry: VirtualFile): DefinesClass =
-    Locate.definesClass(classpathEntry)
+  override def definesClass(classpathEntry: VirtualFile): DefinesClass = {
+    if (classpathEntry.name.toString != "rt.jar")
```

Comment:
This looks a bit unsafe to me.

First, it only matches on the file name, which could break when there are other jars with the same name in the classpath. 

Second, I don't quite understand why this works at all? Newer JDKs (>8) don't provide a `rt.jar` so in Mill we create one dynamically and place it under `~/.mill/ammonite/rt-${jdk-version}.jar`. So, we should never match this correctly, unless there is some other mapping in action. Do you have some explanation for this?

Reply:
Not pretty sure, but this is what `sbt` have done.
https://github.com/sbt/sbt/blob/1.7.x/main/src/main/scala/sbt/internal/VirtualFileValueCache.scala#L29
```scala
apply(converter) { x: VirtualFile =>
      if (x.name.toString != "rt.jar") Locate.definesClass(x)
      else (_: String) => false
    }
```

Reply:
Not if don't have this line, an exception will throw.

Reply:
Thanks for these insight. If `classpathEntry.name` is the full name, this should be rather collision free.